### PR TITLE
EAA: add verdictd's design doc

### PR DESCRIPTION
**Pull request by liangzhou121**
Signed-off-by: Zhou Liang <liang.a.zhou@linux.alibaba.com>

### CODE REVIEW

**Review by jialez0 — APPROVED** [review]

File: eaa/verdictd/docs/design/design.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Introduction
+Cloud computing has become an inevitable trend for consumers, enterprises, and other entities. However, security risks in the cloud are still the #1 barrier for broader cloud adoption.
+
+HW-TEE based cloud security solutions provide a higher level of protection as compared to software-only security measures. 
+There are many HW-TEE security technologies such as Intel SGX, Intel TDX and AMD SEV, which can be used to secure customers' data.
+But customers still encounter following problems during deployment of their containers into HW-TEE based cloud environments.
+- How to identify if the server can be trusted (The target HW-TEE environment wasn't forged by a malicious Hacker)?
+- How to ensure customers' container content wasn't tampered by malicious Hacker before load into HW-TEE based cloud environment?
+- How to ensure the running programs inside target HW-TEE are as expected? Because these programs maybe deploymented by Cloud Service Provider (CSP) and should passed source code's audit.
+
```
Comment (haosanzi) [inline eaa/verdictd/docs/design/design.md:13]: should pass

File: eaa/verdictd/docs/design/design.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+These complex problems can be resloved by HW-TEE's remote attestation technologies. But remote attestation's mechanism and deployment are complicated to general customers.
```
Comment (haosanzi) [inline eaa/verdictd/docs/design/design.md:14]: resloved ---> resolved

File: eaa/verdictd/docs/design/design.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+Enclave Attestation Architecture(EAA) Verdictd was built to simple the deployment of remote attestation procedure. It integrates these different platforms' remote attestations into general service.
```
Comment (haosanzi) [inline eaa/verdictd/docs/design/design.md:15]: simple is adjective , change simple  to simplify

File: eaa/verdictd/docs/design/design.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+In order to maximum Verdictd's scalability, all functionalities were implemented as separately static RUST modules. Each module can be added or excluded during compiling stage.
+
+# Work process
+
+The below diagram depicts the whole of potential working process of EAA.
+
+![workprocess.png](workprocess.png)
+
+- EAA's Attestation Agent(AA) will fetch HW-TEE signed evidence from HW-TEE environment and send to EAA's Verdictd to verify when remote attestation is needed.
+- Verdictd will execute following inspections after receive evidence from AA:
```
Comment (haosanzi) [inline eaa/verdictd/docs/design/design.md:49]: receiving

File: eaa/verdictd/docs/design/design.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+![architecture.png](architecture.png)
+
+The bottom part of the architecture diagram indicates the various HW-TEE environments that are supported by Verdictd.
+It implements a GRPC server that client can use to set Verdictd's configurations and OPA policy files.
+Currently it relies on [rats-tls](https://github.com/alibaba/inclavare-containers/tree/master/rats-tls) to execute remote attestation procedure and build a secure & trusted channel with HW-TEE environment.
+
+## Modular design with scalability
+
+In order to maximum Verdictd's scalability, all functionalities were implemented as separately static RUST modules. Each module can be added or excluded during compiling stage.
```
Comment (haosanzi) [inline eaa/verdictd/docs/design/design.md:40]: maximize

**Comment by haosanzi** [comment]
Please use Grammarly to perform grammar and vocabulary checks

**Comment by liangzhou121** [comment]
> 

Yes, I checked these new contents with Grammarly tool according to your suggestion.

File: eaa/verdictd/docs/design/design.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
+
+Building a trusted, secure, and controlled cloud with hardware root of trust.
+
+# Introduction
```
Comment (jiazhang0) [inline eaa/verdictd/docs/design/design.md:4]: Add an empty line.

File: eaa/verdictd/docs/design/design.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+Cloud computing has become an inevitable trend for consumers, enterprises, and other entities. However, security risks in the cloud are still the #1 barrier for broader cloud adoption.
+
+HW-TEE-based cloud security solutions provide a higher level of protection as compared to software-only security measures.
```
Comment (jiazhang0) [inline eaa/verdictd/docs/design/design.md:7]: Add an empty line.

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

**Review by haosanzi — APPROVED** [review]

**Pull request merged**